### PR TITLE
Payment manager

### PR DESCRIPTION
**Pull request by razmikTovmas**

### CODE REVIEW

File: src/PaymentManager/PaymentManager.Api/Controllers/PayableController.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            get => Path.Combine(_environment.ContentRootPath, "MockData", "PayableMockData.json");
+            set => _ = value;
+        }
+
+        /// <summary>
+        /// This API is created for mocking only, make sure to call it once :)
+        /// </summary>
+        /// <returns></returns>
+        [HttpPost]
+        public async Task Post()
```
Comment (GevorgMartirosyan) [inline src/PaymentManager/PaymentManager.Api/Controllers/PayableController.cs]: Why can't we do this via a migration?

File: src/PaymentManager/PaymentManager.Api/Controllers/ReceivableController.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            get => Path.Combine(_environment.ContentRootPath, "MockData", "ReceivableMockData.json");
+            set => _ = value;
+        }
+
+        /// <summary>
+        /// This API is created for mocking only, make sure to call it once :)
+        /// </summary>
+        /// <returns></returns>
+        [HttpPost]
+        public async Task Post()
```
Comment (GevorgMartirosyan) [inline src/PaymentManager/PaymentManager.Api/Controllers/ReceivableController.cs]: The same here.

File: src/PaymentManager/PaymentManager.Services/Abstractions/Models/PayablePage.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            int totalCount = query.Count();
+            int totalPages = (int)Math.Ceiling(totalCount / (double)pageSize);
+
+            decimal totalPlannedCost = 0;
+            decimal totalActualCost = 0;
+            decimal totalDifference = 0;
+            decimal totalPaid = 0;
+            decimal totalRemaining = 0;
+
+            foreach (var item in query.ToList())
```
Comment (arttonoyan) [inline src/PaymentManager/PaymentManager.Services/Abstractions/Models/PayablePage.cs]: why need execute to list?

File: src/PaymentManager/PaymentManager.Services/Abstractions/Models/PayablePage.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        public int PageSize { get; set; }
+        public int TotalPages { get; set; }
+        public decimal TotalPlannedCost { get; set; }
+        public decimal TotalActualCost { get; set; }
+        public decimal TotalDifference { get; set; }
+        public decimal TotalPaid { get; set; }
+        public decimal TotalRemaining { get; set; }
+        public List<PayableRead> Items { get; set; }
+
+        public static PayablePage GetPayablePage(IQueryable<PayableEntity> query, IMapper mapper, int page, int size)
```
Comment (arttonoyan) [inline src/PaymentManager/PaymentManager.Services/Abstractions/Models/PayablePage.cs]: Move to the Extension method for `IQueryable<PayableEntity>`

File: src/PaymentManager/PaymentManager.Services/Abstractions/Models/ReceivablePage.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    public class ReceivablePage
+    {
+        public int TotalCount { get; set; }
+        public int CurrentPage { get; set; }
+        public int PageSize { get; set; }
+        public int TotalPages { get; set; }
+
+        public List<ReceivableRead> Items { get; set; }
+
+        public static ReceivablePage GetReceivablePage(IQueryable<ReceivableEntity> query, IMapper mapper, int page, int size)
```
Comment (arttonoyan) [inline src/PaymentManager/PaymentManager.Services/Abstractions/Models/ReceivablePage.cs]: Move to Extensions method

File: src/PaymentManager/PaymentManager.Services/Helpers/PayableSearchHelper.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
+﻿using PaymentManager.Repositories.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PaymentManager.Services.Helpers
+{
+    public class PayableSearchHelper : ISearchHelper<PayableEntity>
```
Comment (arttonoyan) [inline src/PaymentManager/PaymentManager.Services/Helpers/PayableSearchHelper.cs]: Move logics like this to the repository.

**Review by arttonoyan — DISMISSED** [review]

**Review by arttonoyan — APPROVED** [review]

**Pull request merged**